### PR TITLE
libx265: fix build to armv8 with apple-clang

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
@jgsogo I'm able to build for Macos M1, could you confirm it works in CI please?

File: recipes/libx265/all/conanfile.py [hunk 1]
```diff
@@ -49,22 +49,23 @@ def config_options(self):
             del self.options.fPIC
         if self.settings.os != "Linux":
             del self.options.with_numa
+        # FIXME: Disable assembly by default if host is arm and compiler apple-clang for the moment.
+        # Indeed, apple-clang is not able to understand some asm instructions of libx265
+        if self.settings.compiler == "apple-clang" and "arm" in self.settings.arch:
```
Comment (SSE4) [inline recipes/libx265/all/conanfile.py:54]: do you have an error? this might not be desired for many people to have assembly optimization disabled...

Comment (SpaceIm) [reply recipes/libx265/all/conanfile.py:54]: Sure, it doesn't compile. But it's better to compile without assembly optimization than nothing? Currently recipe raises if not x86 or x86_64...

**Review by SSE4 — DISMISSED** [review]

**Review by uilianries — DISMISSED** [review]

File: recipes/libx265/all/conanfile.py [hunk 2]
```diff
@@ -87,6 +88,11 @@ def _configure_cmake(self):
             self._cmake.definitions["STATIC_LINK_CRT"] = "T" in str(self.settings.compiler.runtime)
         if self.settings.os == "Linux":
             self._cmake.definitions["PLATFORM_LIBS"] = "dl"
+        # FIXME: too specific, should be handled by CMake helper
+        if tools.is_apple_os(self.settings.os) and self.settings.arch == "armv8":
+            self._cmake.definitions["CMAKE_SYSTEM_PROCESSOR"] = "aarch64"
```
Comment (SpaceIm) [inline recipes/libx265/all/conanfile.py:93]: Maybe it should be:

```suggestion
        if tools.cross_building(self.settings):
            cmake_system_processor = {
                "armv8": "aarch64",
                "armv8.3": "aarch64",
            }.get(str(self.settings.arch), str(self.settings.arch))
            self._cmake.definitions["CMAKE_SYSTEM_PROCESSOR"] = cmake_system_processor
```

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**